src/index.js

```diff
@@ -311,7 +319,8 @@ function strategy (server) {
   return {
     authenticate (request, h) {
       const altAuth = request.query.token ? 'Bearer ' + request.query.token : null;
-      return validate(request.raw.req.headers.authorization || altAuth, h)
+      const sharing = request.query.sharing ? 'Sharing: ' + request.query.sharing : null;
```

Comment:
@tdeleanu I'm not comfortable creating and supporting non-standard authentication mechanism. going this route. Can't we use a regular Bearer token with a "sharing" scope?

Reply:
https://curity.io/resources/learn/token-sharing/#exchanging-a-token

Reply:
@asanchezdelc let's talk this out today. 
This is for allowing a sharing model similar to google drive. 
This is not a new way of authentication.